src/main/java/com/origamisoftware/teach/advanced/services/APIStockServiceAdaptor.java

```
@@ -0,0 +1,62 @@
+package com.origamisoftware.teach.advanced.services;
+
+import com.origamisoftware.teach.advanced.model.StockQuote;
+import com.origamisoftware.teach.advanced.util.Interval;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.List;
+
+public class APIStockServiceAdaptor implements StockService {
+
+
+    private APIService apiQuote;
+
+    public APIStockServiceAdaptor(){
+        this.apiQuote = apiQuote;
+    }
+
+
+    /**
+     * Return the current price for a share of stock  for the given symbol
+     *
+     * @param symbol the stock symbol of the company you want a quote for.
+     *               e.g. APPL for APPLE
+     * @return a  <CODE>BigDecimal</CODE> instance
+     * @throws StockServiceException if using the service generates an exception.
+     *                               If this happens, trying the service may work, depending on the actual cause of the
+     *                               error.
+     */
+    @Override
+    public StockQuote getQuote(String symbol) throws StockServiceException {
+
```

don't return null. 
implement this method 
if you can't implement method throw unsupported operation exception.